nanshe_ipython.ipynb

```diff
@@ -79,6 +75,23 @@
         "html_ext = os.path.extsep + \"html\""
       ]
     },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "import os\n",
+        "from psutil import cpu_count\n",
+        "\n",
+        "cluster_kwargs = {}\n",
+        "client_kwargs = {}\n",
+        "adaptive_kwargs = {\n",
+        "    \"minimum\": 0,\n",
+        "    \"maximum\": int(os.environ.get(\"CORES\", cpu_count())) - 1\n",
```

Comment:
This is taken from [`set_num_workers`]( https://github.com/nanshe-org/nanshe_workflow/blob/v2.11.0/nanshe_workflow/par.py#L62-63 ). `set_num_workers` is designed more around `ipyparallel` where we had to use the CLI and rely on environment variables for information. This is a bit overengineered for Distributed where we can startup the cluster from Python. Plus having this info here makes it more visible and customizable. Hence why this is taken from `set_num_workers` and added here.

`CORES` is mainly used for things like running batch jobs or [testing in the Docker image build]( https://github.com/nanshe-org/docker_nanshe_workflow/blob/91604047aa2128e5ccb87dc38098730c24ac2cf4/Dockerfile#L46 ). Where we need an external way to override the number of CPU cores used and/or `cpu_count` is unreliable.  Otherwise we default to `cpu_count`. We subtract 1 as we need to take into account that 1 core is being used running the Jupyter Notebook and the Distributed Scheduler.